frontend/src/app/api/api/eventTypes.service.ts

```diff
@@ -0,0 +1,286 @@
+            let accessToken = typeof this.configuration.accessToken === 'function'
+                ? this.configuration.accessToken()
+                : this.configuration.accessToken;
+            headers = headers.set('Authorization', 'Bearer ' + accessToken);
+        }
+
+        // to determine the Accept header
+        let httpHeaderAccepts: string[] = [
+            'application/json'
+        ];
+        let httpHeaderAcceptSelected: string = this.configuration.selectHeaderAccept(httpHeaderAccepts);
+        if (httpHeaderAcceptSelected !== undefined) {
+            headers = headers.set('Accept', httpHeaderAcceptSelected);
+        }
+
+        // to determine the Content-Type header
+        let consumes: string[] = [];
+
+        return this.httpClient.get<any>(`${this.basePath}/event_type`,
+            {
+                params: queryParameters,
+                headers: headers,
+                withCredentials: this.configuration.withCredentials,
+            }
+        );
+    }
+
+    /**
+     * Show a event_type
+     * Returns details about a event_type
+     * @param id EventType id to show
+     */
+    public showEventType(id: number): Observable<EventType> {
+        if (id === null || id === undefined) {
+            throw new Error('Required parameter id was null or undefined when calling showEventType.');
+        }
+
+        let headers = this.defaultHeaders;
+
+        // authentication (h2ms_auth) required
+        if (this.configuration.accessToken) {
+            let accessToken = typeof this.configuration.accessToken === 'function'
+                ? this.configuration.accessToken()
+                : this.configuration.accessToken;
+            headers = headers.set('Authorization', 'Bearer ' + accessToken);
+        }
+
+        // to determine the Accept header
+        let httpHeaderAccepts: string[] = [
+            'application/json'
+        ];
+        let httpHeaderAcceptSelected: string = this.configuration.selectHeaderAccept(httpHeaderAccepts);
+        if (httpHeaderAcceptSelected !== undefined) {
+            headers = headers.set('Accept', httpHeaderAcceptSelected);
+        }
+
+        // to determine the Content-Type header
+        let consumes: string[] = [];
+
+        return this.httpClient.get<any>(`${this.basePath}/event_type/${encodeURIComponent(String(id))}`,
+            {
+                headers: headers,
+                withCredentials: this.configuration.withCredentials,
+            }
+        );
+    }
+
+    /**
+     * Update a event_type
+     * Update a event_type in the H2MS application.
+     * @param id EventType id to update
+     * @param body Hopsital object fields that needs to be updated
+     */
+    public updateEventType(id: number, body: EventType): Observable<EventType> {
+        if (id === null || id === undefined) {
+            throw new Error('Required parameter id was null or undefined when calling updateEventType.');
+        }
```

Comment:
You might consider reducing the duplication by turning this into a function.

Maybe adding a checkNotNullOrUndefined(var parameter). [Here's a link to Google Guava's checkNotNull](http://google.github.io/guava/releases/snapshot/api/docs/com/google/common/base/Preconditions.html#checkNotNull-T-)

[Or Java's requireNonNull for inspiration](https://docs.oracle.com/javase/7/docs/api/java/util/Objects.html#requireNonNull(T))